### PR TITLE
Make progress metrics interval configurable

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <ben.ye@bytedance.com>

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

1. Only enable progress metrics if `wait` mode is enabled
2. Make the interval configurable instead of hardcoded 5m
3. Remove the original `progress-metrics` bool flag. Now to disable progress metrics, just set interval to 0.

### CODE REVIEW

File: cmd/thanos/compact.go [hunk 1]
```diff
@@ -694,6 +693,8 @@ func (cc *compactConfig) registerFlag(cmd extkingpin.FlagClause) {
 		Default("5m").DurationVar(&cc.blockViewerSyncBlockTimeout)
 	cmd.Flag("compact.cleanup-interval", "How often we should clean up partially uploaded blocks and blocks with deletion mark in the background when --wait has been enabled. Setting it to \"0s\" disables it - the cleaning will only happen at the end of an iteration.").
 		Default("5m").DurationVar(&cc.cleanupBlocksInterval)
+	cmd.Flag("compact.progress-interval", "How often we should calculate the compaction progress in the background when --wait has been enabled. Setting it to \"0s\" disables it. Now compaction, downsampling and retention progress are supported.").
```
Comment (metonymic-smokey) [inline cmd/thanos/compact.go:696]: Just a small nit. I think phrasing it this way might be better: "frequency of calculating the compaction progress in the background when --wait has been enabled". WDYT?

Comment (yeya24) [reply cmd/thanos/compact.go]: Done.

**Comment by yeya24** [comment]
I'll merge it now as it is a small change.

**Review by GiedriusS — COMMENTED** [review]
:+1:

**Pull request merged**